zerver/lib/user_mutes.py

```diff
@@ -41,3 +42,15 @@ def remove_user_mute(user_profile: UserProfile, muted_user: UserProfile) -> None
 
 def user_is_muted(user_profile: UserProfile, muted_user: UserProfile) -> bool:
     return MutedUser.objects.filter(user_profile=user_profile, muted_user=muted_user).exists()
+
+
+@cache_with_key(get_muting_users_cache_key, timeout=3600 * 24 * 7)
+def get_muting_users(muted_user: UserProfile) -> Set[int]:
+    # This is kind of the inverse of `get_user_mutes` above.
+    # While `get_user_mutes` is mainly used for event system work,
+    # this is used in the message send codepath, to get a list
+    # of IDs of users who have muted a particular user.
+    rows = MutedUser.objects.filter(
+        muted_user=muted_user,
+    ).values("user_profile__id")
+    return {row["user_profile__id"] for row in rows}
```

Comment:
Can you get the raw SQL for this query and do an `EXPLAIN ANALYZE` on it in a `manage.py dbshell` (postgres shell) to make sure the query plan is an index scan?  You should be able to grab the generated SQL via the `queries_captured` context manager and just printing the queries rather than checking the length.